lib/CL/devices/remote/communication.c

```diff
@@ -2034,22 +2035,23 @@ pocl_network_free_device (cl_device_id device)
 // SYNCHRONOUS
 
 cl_int
-pocl_network_create_buffer (remote_device_data_t *ddata, uint32_t mem_id,
-                            uint32_t mem_flags, uint64_t mem_size,
+pocl_network_create_buffer (remote_device_data_t *ddata, cl_mem mem,
```

Comment:
Since we are passing a whole cl_mem object now, it may be a good idea to assert that at least the most critical fields have been set to some sensible values. Either here or in remote.c.

Also CL_MEM_USES_SVM_POINTER is mentioned in messages.h as required if host_ptr is to be set, so that would be good to assert here as well.